packages/codeeditor/src/viewer.ts

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) Jupyter Development Team.
+// Distributed under the terms of the Modified BSD License.
+
+import { StackedLayout, Widget } from '@lumino/widgets';
+import { CodeEditor, CodeEditorWrapper } from './';
+
+export class CodeViewerWidget extends Widget {
```

Comment:
I saw that FileEditor has some logic to ensure focus on `mousedown`. Do you think it is needed here?

Reply:
It is not needed here since this is readonly. That functionality makes sure that the editor get keyboard focus for typing, which you can't doing in readonly